src/StringParser.php

```diff
@@ -26,4 +26,17 @@ public function required()
     {
         return parent::required();
     }
+
+    /**
+     * @param string $defaultValue
+     * @return string
+     */
+    public function defaultsToIfEmpty($defaultValue)
+    {
+        if ($this->value === '') {
```

Comment:
One question, could this be placed in `AbstractValueParser `? This way you have all types covered. And perhaps `empty()` would be sufficient instead of `=== ''`?
```php
$emptyString = '';
$isEmpty = empty($emptyString);
var_dump($isEmpty); // bool(true)
```